### DISCUSSION TITLE
entityManager no realiza INSERTS en los tests

### DISCUSSION THREAD

**Discussion by 23737k** [dds-jv/foro#313] [Consultas de persistencia] [answered]
Hola. Tengo un problemilla con los tests. Ya mapeé todas las clases (anotations, persistence.xml) y cuando quiero correr el test LugarTest, me da todo en verde menos el de puedeRecuperarLugares(). Revisé y ví que el entity sólo hace un select. Intenté con flush() pero me tira TransactionRequiredException. Ni idea por qué ocurre, digo.. no debería dar ese error ya que el test hereda de AbstractPersistenceTest que se supone que ejecuta una transacción. 

Pd: ya revisé como loco StackOverFlow y no encontré solución. 
Gracias.

**_LugarTest dando failure y el entity hace sólo un SELECT_**

![image](https://user-images.githubusercontent.com/82678482/134085084-8bbc57b1-24a6-4fcb-8c7d-419fc796bbe3.png)
![image](https://user-images.githubusercontent.com/82678482/134085164-599345cf-09de-4b58-bd9b-062807a53227.png)

**_Usando el flush()_**
![image](https://user-images.githubusercontent.com/82678482/134085656-15dd8519-60d9-4c34-b261-b99878357873.png)

**Comment by AlejandroDeheza** [comment, marked as answer]
Buenas, estuve varios dias con este problema tambien, y lo arregle de casualidad agregando esto en todos los archivos de tests

~~~
  @BeforeEach
  public void antes() {
    this.beginTransaction();
  }

  @AfterEach
  public void despues() {
    this.rollbackTransaction();
  }
~~~

No se esactamente porque es esto, pero supongo que es porque el AbstractPersistenceTest tiene anotaciones de JUnit 4 en ves de JUnit 5, que se pueden ver viendo su codigo. O sea esto que agregue, en realidad esta en AbstractPersistenceTest pero con anotaciones de la version anterior de JUnit (suponiendo que tenes JUnit 5 en el POM).

Despues decime si te funciono. Si algun ayudante/profe sabe mas sobre esto y nos comenta como es la cosa seria un golazo

Saludos!

**Comment by 23737k** [comment]
¡Genio! Muchas gracias, me estaba rompiendo el coco con esto. Igual me da curiosidad por qué pasa esto. Éxitos.

**Reply by gastonprieto** [reply]
Genial que lo vieron !!! Entiendo que el problema es porque el runner de Junit 5 no reconoce las annotations de junit 4

**Comment by tomaslcarballo** [comment]
Genio! Yo también estuve con el mismo problema y ya funciona bien con lo que comentaste. 
Muchas gracias!!
